ambari/datadog_checks/ambari/ambari.py

```diff
@@ -0,0 +1,198 @@
+# (C) Datadog, Inc. 2019
+# All rights reserved
+# Licensed under a 3-clause BSD style license (see LICENSE)
+from six import iteritems
+from requests.auth import HTTPBasicAuth
+import requests
+
+
+from datadog_checks.base import AgentCheck
+from datadog_checks.base.errors import CheckException
+from . import common
+
+# Tag templates
+CLUSTER_TAG_TEMPLATE = "ambari_cluster:{}"
+HOST_TAG = "ambari_host:"
+SERVICE_TAG = "ambari_service:"
+COMPONENT_TAG = "ambari_component:"
+
+# URL queries
+COMPONENT_METRICS_QUERY = "/components?fields=metrics"
+SERVICE_INFO_QUERY = "?fields=ServiceInfo"
+
+# Response fields
+METRICS_FIELD = "metrics"
+
+
+class AmbariCheck(AgentCheck):
+    def __init__(self, *args, **kwargs):
+        super(AmbariCheck, self).__init__(args, kwargs)
+        self.hosts = []
+        self.clusters = []
+        self.services = []
+
+    def check(self, instance):
+        server = instance.get("url", "")
+        port = str(instance.get("port", ""))
+        path = str(instance.get("path", ""))
+        base_tags = instance.get("tags", [])
+        whitelisted_services = instance.get("services", [])
+        authentication = {'username': instance.get("username", ""),
+                          'password': instance.get("password", "")
+                          }
+        whitelisted_metrics = [str(h) for h in instance.get("metric_headers", [])]
+
+        base_url = "{}:{}{}".format(server, port, path)
+        clusters = self.get_clusters(base_url, authentication)
+        self.get_host_metrics(base_url, authentication, clusters, base_tags)
+        self.get_service_metrics(base_url, authentication, clusters, whitelisted_services, whitelisted_metrics, base_tags)
+
+    def get_clusters(self, base_url, authentication):
+        clusters_endpoint = common.CLUSTERS_URL.format(base_url=base_url)
+
+        resp = self._make_request(clusters_endpoint, authentication)
+        if resp is None:
+            self._submit_service_checks("can_connect", self.CRITICAL, ["url:{}".format(base_url)])
+            raise CheckException(
+                "Couldn't connect to URL: {}. Please verify the address is reachable".format(clusters_endpoint))
+
+        self._submit_service_checks("can_connect", self.OK, ["url:{}".format(base_url)])
+        return [cluster.get('Clusters').get('cluster_name') for cluster in resp.get('items')]
+
+    def get_host_metrics(self, base_url, authentication, clusters, base_tags):
+        for cluster in clusters:
+            cluster_tag = CLUSTER_TAG_TEMPLATE.format(cluster)
+            hosts_list = self._get_hosts_info(base_url, authentication, cluster)
+
+            for host in hosts_list:
+                if host.get(METRICS_FIELD) is None:
+                    self.log.warning("No metrics received for host {}".format(host.get('Hosts').get('host_name')))
+                    continue
+
+                metrics = self.flatten_host_metrics(host.get(METRICS_FIELD))
```

Comment:
Please do `host_metric = host.get(METRICS_FIELD)`above, and reuse it the 2 times you use it.